### ISSUE TITLE
Sudden crash on Android using Flutter shakebugs 17.0.1 

### ISSUE THREAD

**Issue by guixijing** [shakebugs/shake-flutter#8]
After we updated the Flutter Shakebugs SDK to version 17.0.1, we encounter this error every week.

<img width="808" alt="image" src="https://github.com/user-attachments/assets/76f42827-6545-4aa0-a2a0-c041c9a8c687">

I believe this issue occurs because the previous activity hasn't been unregistered successfully. Could you please help us resolve this issue, as it results in a spike in crashes each week? Thanks a lot!

**Comment by alicja-ogonowska** [comment]
Experiencing the same issue with 17.0.1 and 17.0.0. Downgrade to 16.2.1 helped.
Crashes occur when users try to take a screenshot.

```
   Fatal Exception: java.lang.RuntimeException: Unable to resume activity {com.olx.realestate.app.imovirtual/com.olx.realestate.app.MainActivity}: java.lang.IllegalStateException: Capture observer already registered with the activity
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5244)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5277)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
       at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:60)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:282)
       at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:150)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2595)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loopOnce(Looper.java:232)
       at android.os.Looper.loop(Looper.java:317)
       at android.app.ActivityThread.main(ActivityThread.java:8592)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
```

**Comment by mstanic-shake** [comment]
This should be resolved with the latest version. Please update a version and let us know if you still have a problem

**Comment by alicja-ogonowska** [comment]
@mstanic-shake which latest version do you mean? Latest version I see on pub.dev is 17.0.1, which is exactly the version that this issue was reported for.

**Comment by guixijing** [comment]
@alicja-ogonowska I think they have resolved it with android dependency [17.0.2](https://central.sonatype.com/artifact/com.shakebugs/shake/17.0.2). The new build after 11th Sep 2024 would have the fix since Flutter 17.0.1 depends on [native Android Shakebugs 17.0.+](https://github.com/shakebugs/shake-flutter/blob/3d756626c79bf358210f0ca833d22d640e0f03dc/android/build.gradle#L53). No update required if you are using shake-flutter 17.0.1. So far no more spike of crashes for us.
